components/portal/messages/services.js

```diff
@@ -208,6 +208,61 @@ define(['angular'], function(angular) {
         };
 
         /**
+         * Filter the array of messages based on if
+         * a url to get the title was requested
+         * @param {Array} messages An array of message objects
+         * @return {filteredMessages[]} an array of messages that
+         *   includes messages that requested a title from a url
+         */
+        var getMessagesByTitle = function(messages) {
+          var promises = [];
+          var filteredMessages = [];
+          var toBeDiscarded = [];
+          angular.forEach(messages, function(message) {
+            if (message.titleUrl) {
+              promises.push($http.get(message.titleUrl)
+                .then(function(result) {
+                  if (result.data) {
+                    var titleObject = result.data;
+                    if (titleObject.result && titleObject.result.length > 0) {
+                      var fromApi = titleObject.result[0];
+                      if (fromApi.full) {
```

Comment:
I'm not fond of the idea of four nested `if`s, but I see why there are here. :white_check_mark: 
If, in the future, `lodash` were added `lodash.get` could abstract some of this complexity away.

Reply:
We're hoping to move most of this filtering logic to a microservice in the near-ish future, so hopefully the front end won't have to bother with it at all.